python/pyspark/sql/functions.py

```diff
@@ -1756,6 +1756,7 @@ def __call__(self, *cols):
 @since(1.3)
 def udf(f, returnType=StringType()):
     """Creates a :class:`Column` expression representing a user defined function (UDF).
+    Note that the user-defined functions should be deterministic.
```

Comment:
Thanks for doing this!  I would say `must be` deterministic.  We might even want to say that "due to optimization duplicate invocations may be eliminated or the function may even be invoked more times than it is present in the query".